agent/lib/kontena/workers/log_worker.rb

```diff
@@ -51,31 +65,53 @@ def queue_processing?
 
     def start
       sleep 1 until Actor[:etcd_launcher].running?
+      start_container_workers
+      async.process_queue
+    end
+
+    def start_container_workers
       Docker::Container.all.each do |container|
         begin
           self.stream_container_logs(container) unless container.skip_logs?
         rescue Docker::Error::NotFoundError
           # Could be thrown since container.skip_logs? actually loads the container details
         end
       end
-
-      async.process_queue
     end
 
     def process_queue
+      current_actor = Actor.current
```

Comment:
This is going to be several (~5?) sync thread-to-actor calls for **each** log line, with all of the associated overhead. The `defer` does not smell good...